pkg/recommender/metrics/mock_prometheus_api.go

```diff
@@ -0,0 +1,76 @@
+package metrics
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"time"
+)
+
+type MockApi struct{}
```

Comment:
don't export it in application code. This should be moved to `prometheus_provider_mock_test.go` to be invisible outside of tests.

Reply:
Changed it.
`type mockApi struct {
	q  func(query string, ts time.Time) (model.Value, v1.Warnings, error)
	qr func(query string, r v1.Range) (model.Value, v1.Warnings, error)
}`
Only mockApi constructor is exported in metrics package for testing purposes.